### PR TITLE
Personal Pricing: Added validation on income so that it can only be an integer

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
- Fixes https://github.com/mitodl/micromasters/issues/2470#issuecomment-279479984

#### What's this PR do?
- Throws error when prices entered on 

#### How should this be manually tested?
(Required)
- Open dashboard and then `Cost Calculator`
 - Try to input string, alpha numeric
 - Try to enter decimal number
 - Try to enter special character 
 - Try to enter whole numbers

screen shot 2017-02-14 at 8 09 17 pm

screen shot 2017-02-14 at 8 12 35 pm

### CODE REVIEW

File: static/js/containers/FinancialAidCalculator.js [hunk 1]
```diff
@@ -56,7 +55,7 @@ const currencySelect = (update, current) => (
 
 const salaryUpdate = R.curry((update, current, e) => {
   let newEdit = R.clone(current);
-  newEdit.income = sanitizeNumberString(20, e.target.value);
+  newEdit.income = e.target.value
```
Comment (aliceriot) [inline static/js/containers/FinancialAidCalculator.js]: why not use the `filterPositiveInt` function we define in `util/util.js`? then we can keep the in-line behavior, preventing the user from typing a decimal in the first place.

you could do something like

```js
const sanitizeIncome = R.compose(filterPositiveInt, sanitizeNumberString(20));

newEdit.income = sanitizeIncome(e.target.value);
```

Comment (amir-qayyum-khan) [inline static/js/containers/FinancialAidCalculator.js]: this wont allow user to enter (.) i.e 2000.00 and see error message.
it changes 2000.00 to 200000. try ur self, copy 2000.00  and paste in field you will see it change into 200000

Comment (aliceriot) [inline static/js/containers/FinancialAidCalculator.js]: Oh, that makes sense. How about we add the error message but keep the `sanitizeNumberString` call?

Comment (amir-qayyum-khan) [inline static/js/containers/FinancialAidCalculator.js]: how? i think `newEdit.income` is what trigger validation/

Comment (aliceriot) [inline static/js/containers/FinancialAidCalculator.js]: just undo the change you made to line 59 in this file, keep it as-is on master. `sanitizeNumberString` basically just removes leading zeros.

Comment (amir-qayyum-khan) [inline static/js/containers/FinancialAidCalculator.js]: - It dont allow me to enter text and neither show the validation (please enter whole number)
- it dont allow me to add `.xy`
- it when you copy `555.50` into income it change it to `55550`

that the is reason @pdpinch opened this issue

Comment (aliceriot) [inline static/js/containers/FinancialAidCalculator.js]: ah ok now I see the issue. sorry for the lack of clarity! I wasn't following your explanations.

File: static/js/lib/validation/profile.js [hunk 2]
```diff
@@ -299,5 +299,10 @@ export const validateFinancialAid = (edit: FinancialAidState): FinancialAidValid
   if (!edit.checkBox) {
     errors['checkBox'] = 'You must agree to these terms';
   }
+
+  if (edit.income && !/^\d+$/.test(edit.income)) {
```
Comment (aliceriot) [inline static/js/lib/validation/profile.js]: I'm not sure this is exactly the regex we want - I think it might be more clear to test for non-digit characters, instead of `!`-ing a test for all digits.

so maybe something like `/\D+/` ?

Comment (aliceriot) [inline static/js/lib/validation/profile.js]: `+` isn't even really necessary, I think `/\D/` would do the job.

Comment (amir-qayyum-khan) [inline static/js/lib/validation/profile.js]: @aliceriot `!/^\d+$/.test(edit.income)` this give us all string which are not number, including `dot(.)` special characters.

Otherwise we have to cater those in our regex

Comment (aliceriot) [inline static/js/lib/validation/profile.js]: will `/\D/` not do that?

Comment (aliceriot) [inline static/js/lib/validation/profile.js]: `\D` matches any non-digit character

Comment (amir-qayyum-khan) [inline static/js/lib/validation/profile.js]: nop,

Comment (aliceriot) [inline static/js/lib/validation/profile.js]: I don't think you don't need the quantifiers or the anchors (`^`, `+`, `$`) either because all we care about is whether there is a non-digit character anywhere in the string

Comment (aliceriot) [inline static/js/lib/validation/profile.js]: `!/^\d+$/.test(edit.income)` and `/\D/.test(edit.income)` are, as far as I can tell, totally equivalent - I'm just advocating for the latter because the regular expression is simpler.

Comment (amir-qayyum-khan) [inline static/js/lib/validation/profile.js]: you r right fixed

File: static/js/lib/validation/profile.js [hunk 3]
```diff
@@ -299,5 +299,10 @@ export const validateFinancialAid = (edit: FinancialAidState): FinancialAidValid
   if (!edit.checkBox) {
     errors['checkBox'] = 'You must agree to these terms';
   }
+
+  if (edit.income && /\D+/.test(edit.income)) {
+    errors['income'] = 'Please only use whole numbers.';
+  }
+
```
Comment (aliceriot) [inline static/js/lib/validation/profile.js]: could you add a quick test of this behavior in the `validation/profile_test.js` file?

Comment (amir-qayyum-khan) [inline static/js/lib/validation/profile.js]: Done @aliceriot
